tests/integration/shared-libs/cht-datasource/contact.spec.js

```diff
@@ -0,0 +1,381 @@
+const utils = require('@utils');
+const personFactory = require('@factories/cht/contacts/person');
+const placeFactory = require('@factories/cht/contacts/place');
+const userFactory = require('@factories/cht/users/users');
+const {getRemoteDataContext, Qualifier, Contact} = require('@medic/cht-datasource');
+const {expect} = require('chai');
+const {setAuth, removeAuth} = require('./auth');
+
+describe('cht-datasource Contact', () => {
+  // NOTE: this is a common word added to contacts to fetch them
+  const commonWord = 'contact';
+  // NOTE: this is a search word added to contacts for searching purposes
+  // the value was chosen such that it is a sub-string of the short_name which
+  // gives double output from the couchdb view
+  const searchWord = 'freetext';
+  // the fields `search` and `short_name` exist for the unique search by freetext based searching
+  // whereas the `name` field is for just simple searching
+  // combining them to have similar text is not done here because the order in which the docs
+  // were being returned were not consistent, meaning the order could be [contact0, contact1, contact2]
+  // in the first run whereas another in another giving a non-consistent expected value to match against
+  const contact0 = utils.deepFreeze(personFactory.build({
+    name: 'contact0', role: 'chw', notes: searchWord, short_name: searchWord + '0'
+  }));
+  const contact1 = utils.deepFreeze(personFactory.build({
+    name: 'contact1', role: 'chw_supervisor', notes: searchWord, short_name: searchWord + '1'
+  }));
+  const contact2 = utils.deepFreeze(personFactory.build({
+    name: 'contact2', role: 'program_officer', notes: searchWord, short_name: searchWord + '2'
+  }));
+  const placeMap = utils.deepFreeze(placeFactory.generateHierarchy());
+  const place1 = utils.deepFreeze({
+    ...placeMap.get('health_center'),
+    contact: {_id: contact1._id},
```

Comment:
kind of a shame that `generateHierarchy` doesn't accept any kind of parameters that would allow adding additional fields to the created places.